src/main/java/de/tum/in/www1/artemis/domain/notification/GroupNotificationFactory.java

```diff
@@ -165,59 +199,100 @@
     public static GroupNotification createNotification(Post post, User author, GroupNotificationType groupNotificationType, NotificationType notificationType, Course course) {
         String title;
         String text;
+        String[] placeholderValues;
         GroupNotification notification;
         switch (notificationType) {
             case NEW_EXERCISE_POST -> {
                 Exercise exercise = post.getExercise();
                 title = NEW_EXERCISE_POST_TITLE;
-                text = "Exercise \"" + exercise.getTitle() + "\" got a new post.";
-                notification = new GroupNotification(course, title, text, author, groupNotificationType);
+                text = NEW_EXERCISE_POST_TEXT;
+                placeholderValues = ArrayUtils.addAll(generatePlaceholderValuesForMessageNotifications(course, post), exercise.getTitle());
```

Comment:
Improvement suggestion: Extract the placeholderValues assignment to after the switch statement. In the different cases, just define the tail. 

```java
class GroupNotificationService {
    GroupNotification createNotification(Post post, User author, GroupNotificationType groupNotificationType, NotificationType notificationType, Course course) {
        String title;
        String text;
        List<String> tail;
        String target;

        switch (notificationType) {
            case NEW_LECTURE_POST -> {
                Lecture lecture = post.getLecture();
                title = NEW_LECTURE_POST_TITLE;
                text = NEW_LECTURE_POST_TEXT;
                tail = List.of(lecture.getTitle());
                target = createLecturePostTarget(post, course);
            }
        }

        placeholderValues = ArrayUtils.addAll(generatePlaceholderValuesForMessageNotifications(course, post), lecture.getTitle());


        notification = new GroupNotification(course, title, text, true, placeholderValues, author, groupNotificationType);
        notification.setTransientAndStringTarget(createLecturePostTarget(post, course));
    }
}
```

Reply:
I refactored this.